packages/mobx/src/types/observableannotation.ts

```diff
@@ -38,6 +46,71 @@ function extend_(
     )
 }
 
+function decorate_20223_(
+    this: Annotation,
+    desc,
+    context: ClassAccessorDecoratorContext | ClassFieldDecoratorContext
+) {
+    if (__DEV__) {
+        assert20223DecoratorType(context, ["accessor", "field"])
+    }
+
+    const ann = this
+    const { kind, name, addInitializer } = context
+
+    // Backwards/Legacy behavior, expects makeObservable(this)
+    if (kind == "field") {
+        addInitializer(function () {
+            storeAnnotation(this, name, ann)
+        })
+        return
+    }
+
+    // The laziness here is not ideal... It's a workaround to how 2022.3 Decorators are implemented:
+    //   `addInitializer` callbacks are executed _before_ any accessors are defined (instead of the ideal-for-us right after each).
+    //   This means that, if we were to do our stuff in an `addInitializer`, we'd attempt to read a private slot
+    //   before it has been initialized. The runtime doesn't like that and throws a `Cannot read private member
+    //   from an object whose class did not declare it` error.
+    const initializedObjects = new WeakSet()
+
+    function initializeObservable(target, value) {
+        const adm: ObservableObjectAdministration = asObservableObject(target)[$mobx]
+        const observable = new ObservableValue(
+            value,
+            ann.options_?.enhancer ?? deepEnhancer,
+            __DEV__ ? `${adm.name_}.${name.toString()}` : `ObservableObject.${name.toString()}`,
+            false
+        )
+        adm.values_.set(name, observable)
+        initializedObjects.add(target)
```

Comment:
`asObservableObject` is idempotent, so I'm not 100% we need the `initializedObjects`? This feels a bit irky, but probably that can possibly be optimised later, fine for now I guess. I imagine we could do a check with `asObservableObject(target)[$mobx].values_.has` if we still need to init.

Reply:
That works too. This approach mimics how TS handles some of the `accessor` stuff in the transpiled code.

Reply:
I'm crossing my fingers! That changing would make this so much cleaner (and probably a little faster since we'll avoid the `has()` call